docs/proposals/logging_rearch.md

```diff
@@ -0,0 +1,55 @@
+# Re architect origin-aggregated-logging to use updated API objects and Templates to install
+
+## Why?
+
+Currently when we develop features or make changes that require configuration
+updates as well as updates to our images we must coordinate those changes in two
+repositories which can lead to things being out of sync or missed.
+
+We should also take the time to evaluate how we deploy/configure the logging
+stack and incorporate newer API objects where able, such as using Stateful Sets
+to deploy Elasticsearch instead of Deployment Configs.
+
+## How?
+
+Taking from our experiences with supporting operations and customers, we should
+be sure to incorporate those gotchas while minimizing the number of hacky
+approaches we've included over releases.
+
+We will need to create templates that allow granular deployment options (e.g.
+how do we deploy Elasticsearch with stateful sets so that we can have different
+zones and some be masters) while still maintaining the fewest number of template
+files to keep the role simple.
```

Comment:
in my PR https://github.com/openshift/openshift-ansible/pull/5001 I have logic of deployment of certain node type in `tasks/node_provision.yaml`, so it should be relatively  easy to abstract to use OCP template for "similarly-purposed" nodes (i.e. for masters, or for clients or for data nodes).